test/persistence-hooks.suite.js

```diff
@@ -3240,13 +3249,17 @@ module.exports = function(dataSource, should, connectorCapabilities) {
           { id: existingInstance.id, name: 'updated name' },
           function(err, instance) {
             if (err) return done(err);
-            ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+            var expectedContext = aCtxForModel(TestModel, {
               data: {
                 id: existingInstance.id,
                 name: 'updated name',
               },
-              isNewInstance: false,
-            }));
+            });
+
+            if (dataSource.connector.upsertWithWhere) {
+              expectedContext.isNewInstance = false;
```

Comment:
Interesting. I would expect the optimized version to have less information than the unoptimized version, simply because the unoptimized version is invoking primitive operations (find, create new instance or update existing instance) as opposed to the optimized version that may not know in advance whether an update or an create will be performed by the backing datasource.

IIRC, we use this pattern in different places:

```
if (dataSource.connector.<atomicOperation>)
  // expect isNewInstance is undefined
else
  // expect isNewInstance is true of false
```

Could you please explain why this case is different?